### ISSUE TITLE
Setting custom parameters programmatically does not work

### ISSUE THREAD

**Issue by fry69** [nalaso/anthropic-vertex-ai#5]
When this provider gets evaluated `createAnthropicVertex();` gets called at the [end](https://github.com/nalaso/anthropic-vertex-ai/blob/main/src/anthropic-vertex-provider.ts#L160).

If you follow the coe path, the default `anthropicVertex` provider then cannot get created with an empty `options` property, because `loadSetting` gets called with environment variables set and will complain that those variables do not get return a valid value, after falling through, see here -> https://github.com/vercel/ai/blob/main/packages/provider-utils/src/load-setting.ts#L42

This produces an error, even if `createAnthropicVertex()` gets called with a correct `options` property later, practically excluding this method to set `region`, `projectID` and other options programmatically.

It took me a while to trace the following error message back to the line mentioned above ->
```
/Users/fry/code-local/ai/node_modules/.pnpm/@ai-sdk+provider-utils@1.0.16_zod@3.23.8/node_modules/@ai-sdk/provider-utils/src/load-setting.ts:45
    throw new LoadSettingError({
          ^


LoadSettingError [AI_LoadSettingError]: Google Vertex project id setting is missing. Pass it using the 'projectId' parameter or the GOOGLE_VERTEX_PROJECT_ID environment variable.
    at loadSetting (/Users/fry/code-local/ai/node_modules/.pnpm/@ai-sdk+provider-utils@1.0.16_zod@3.23.8/node_modules/@ai-sdk/provider-utils/src/load-setting.ts:45:11)
    at createAnthropicVertex (/Users/fry/code-local/ai/packages/anthropic-vertex/src/anthropic-vertex-provider.ts:91:16)
    at <anonymous> (/Users/fry/code-local/ai/packages/anthropic-vertex/src/anthropic-vertex-provider.ts:177:32)
    at ModuleJob.run (node:internal/modules/esm/module_job:262:25)
    at async onImport.tracePromise.__proto__ (node:internal/modules/esm/loader:482:26)
    at async asyncRunEntryPointWithESMLoader (node:internal/modules/run_main:117:5) {
  cause: undefined,
  [Symbol(vercel.ai.error)]: true,
  [Symbol(vercel.ai.error.AI_LoadSettingError)]: true
}
```

The same bug seems also be present in the original `GoogleVertex` provider -> https://github.com/vercel/ai/blob/main/packages/google-vertex/src/google-vertex-provider.ts#L112

Maybe the weird initialization code in that provider works around this bug somehow, but that is beyond my comprehension.

---

TLDR; If I comment [this line](https://github.com/nalaso/anthropic-vertex-ai/blob/main/src/anthropic-vertex-provider.ts#L160) out, this code works ->
```ts
import { createAnthropicVertex } from "@ai-sdk/anthropic-vertex";
import { generateText } from "ai";

const vertexProvider = createAnthropicVertex({
  region: 'europe-west1',
  projectId: 'my-project-1234',
});

const vertexHaiku = vertexProvider('claude-3-haiku@20240307');

const { text } = await generateText({
  model: vertexHaiku,
  prompt: 'What is 6x12?',
});

console.log(text);
```
When I leave the line in, I get above error.

---

**Comment by nalaso** [comment]
It’s working on my end. Let me check again. Also it’s `anthropic-vertex-ai`, not @ai-sdk/anthropic-vertex.

**Comment by fry69** [comment]
It is definitely not working without setting environment variables. I explained in detail why.

You are correct that this example still has the wrong import from my early test when I started to use your pull request (and `pnpm link` to that inside the local `ai` repository), it does not change the facts though.

Here is my current test script ->
```ts
import { createAnthropicVertex } from 'anthropic-vertex-ai';
import { generateText, streamText } from 'ai';

const vertexProvider = createAnthropicVertex({
  region: 'europe-west1',
  projectId: 'my-id-1234',
});
const vertexSonnet = vertexProvider('claude-3-5-sonnet@20240620');
const vertexHaiku = vertexProvider('claude-3-haiku@20240307');

const { text } = await generateText({
  model: vertexHaiku,
  prompt: 'What is 6x12?',
});

console.log(text);

const { textStream } = await streamText({
  model: vertexSonnet,
  prompt: 'Write a long poem about embedding models.',
});

for await (const textPart of textStream) {
  console.log(textPart);
}
```
which produces this error without setting environment variables ->
```
14 |   constructor({
15 |     name: name13,
16 |     message,
17 |     cause
18 |   }) {
19 |     super(message);
         ^
AI_LoadSettingError: Google Vertex project id setting is missing. Pass it using the 'projectId' parameter or the GOOGLE_VERTEX_PROJECT_ID environment variable.
```
If you follow the code path as I described you can see why this is happening.

**Comment by nalaso** [comment]
issue has been solved in v1.0.0
